allenact_plugins/manipulathor_plugin/manipulathor_environment.py

```diff
@@ -0,0 +1,407 @@
+            **kwargs,
+    ) -> None:
+        """Starts the ai2thor controller if it was previously stopped.
+
+        After starting, `reset` will be called with the scene name and move magnitude.
+
+        # Parameters
+
+        scene_name : The scene to load.
+        move_mag : The amount of distance the agent moves in a single `MoveAhead` step.
+        kwargs : additional kwargs, passed to reset.
+        """
+        if self._started:
+            raise RuntimeError(
+                "Trying to start the environment but it is already started."
+            )
+
+        self.controller = self.create_controller()
+
+        if (
+                self._start_player_screen_height,
+                self._start_player_screen_width,
+        ) != self.current_frame.shape[:2]:
+            self.controller.step(
+                {
+                    "action": "ChangeResolution",
+                    "x": self._start_player_screen_width,
+                    "y": self._start_player_screen_height,
+                }
+            )
+
+        self._started = True
+        self.reset(scene_name=scene_name, move_mag=move_mag, **kwargs)
+
+    def reset(
+            self,
+            scene_name: Optional[str],
+            move_mag: float = 0.25,
+            **kwargs,
+    ):
+        self._move_mag = move_mag
+        self._grid_size = self._move_mag
+
+        if scene_name is None:
+            scene_name = self.controller.last_event.metadata["sceneName"]
+        # self.reset_init_params()#**kwargs) removing this fixes one of the crashing problem
+
+        # to solve the crash issue
+        # TODO why do we still have this crashing problem?
+        try:
+            reset_environment_and_additional_commands(self.controller, scene_name)
+        except Exception as e:
+            print("RESETTING THE SCENE,", scene_name, 'because of', str(e))
+            self.controller = ai2thor.controller.Controller(
+                **self.env_args
+            )
+            reset_environment_and_additional_commands(self.controller, scene_name)
+
+        if self.object_open_speed != 1.0:
+            self.controller.step(
+                {"action": "ChangeOpenSpeed", "x": self.object_open_speed}
+            )
+
+        self._initially_reachable_points = None
+        self._initially_reachable_points_set = None
+        self.controller.step({"action": "GetReachablePositions"})
+        if not self.controller.last_event.metadata["lastActionSuccess"]:
+            warnings.warn(
+                "Error when getting reachable points: {}".format(
+                    self.controller.last_event.metadata["errorMessage"]
+                )
+            )
+        self._initially_reachable_points = self.last_action_return
+
+        self.list_of_actions_so_far = []
+
+    def randomize_agent_location(
+            self, seed: int = None, partial_position: Optional[Dict[str, float]] = None
+    ) -> Dict:
+        raise Exception("not used")
```

Comment:
Why explicitly remove this? I'd also change this exception to a `NotImplementedError`.

Reply:
randomizing agent is done differently than the other frameworks, this is due to the collision of the arm during agent teleport which needs to be taken care of. That's why I prefer to make sure users don't use this function to get agent locations.